examples/tokens/ERC1155ownable.vy

```diff
@@ -0,0 +1,358 @@
+
+# NFT marketplace compatibility
+name: public(String[1024])
+symbol: public(String[1024])
+
+# Interface IDs
+ERC165_INTERFACE_ID: constant(bytes4)  = 0x01ffc9a7
+ERC1155_INTERFACE_ID: constant(bytes4) = 0xd9b67a26
+ERC1155_INTERFACE_ID_METADATA: constant(bytes4) = 0x0e89341c
+
+# mappings
+supportsInterfaces: constant(bytes4[3]) = [ERC165_INTERFACE_ID, ERC1155_INTERFACE_ID, ERC1155_INTERFACE_ID_METADATA]
+
+# Mapping from token ID to account balances
+balances: HashMap[uint256,HashMap[address, uint256]]            
+
+# Mapping from account to operator approvals
+operatorApprovals: HashMap[address, HashMap[address, bool]]     
+
+############### events ###############
+event Paused:
+    # @dev Emits a pause event with the address that paused the contract
+    account: address
+
+event unPaused:
+    # @dev Emits an unpause event with the address that paused the contract
+    account: address
+
+event OwnershipTransferred:
+    # @dev Emits smart contract ownership transfer
+    # @param previousOwner The current, soon to be previous owner of the contract
+    # @param newOwner The new owner of the contract
+    previouwOwner: address 
+    newOwner: address
+
+event TransferSingle:
+    # @dev Emits on transfer of a single token
+    # @param operator the address initiating the transfer. owner or authorized operator
+    # @param fromAddress the address sending the token
+    # @param to the address receiving the token
+    # @param id the token ID
+    # @param value the amount of tokens
+    operator:   indexed(address)
+    fromAddress: indexed(address)
+    to: indexed(address)
+    id: uint256
+    value: uint256
+
+event TransferBatch:
+    # @dev Emits on batch transfer of tokens. the ids array correspond with the values array by their position
+    # @param operator the address initiating the transfer. owner or authorized operator
+    # @param fromAddress the address sending the token
+    # @param to the address receiving the token
+    # @param ids list of token ids
+    # @param values list of amounts for each token 
+    operator: indexed(address) # indexed
+    fromAddress: indexed(address)
+    to: indexed(address)
+    ids: DynArray[uint256, BATCH_SIZE]
+    values: DynArray[uint256, BATCH_SIZE]
+
+event ApprovalForAll:
+    # @dev This emits when an operator is enabled or disabled for an owner. The operator can manage
+    # @dev all NFTs of the owner.
+    # @param account Owner of NFT.
+    # @param operator Address to which we are setting operator rights.
+    # @param approved Status of operator rights(true if operator rights are given and false if revoked).
+    account: indexed(address)
+    operator: indexed(address)
+    approved: bool
+
+############### interfaces ###############
+implements: ERC165
+
+interface IERC1155Receiver:
+    def onERC1155Received(operator: address, fromAddress: address, to: address, id: uint256, _value: uint256, data: bytes32) -> bytes32: payable
```

Comment:
Your change is missing the to/receiver address I believe. Will change this with sender / receiver as new names

Reply:
not sure about the CALLBACK_NUMBYTES.
Do you mean add `CALL_BACK_NUMBYTES: constant(Bytes4096])` as a contract constant?

Reply:
> Your change is missing the to/receiver address I believe. Will change this with sender / receiver as new names

That argument isn't in the spec, I think because the callback is directly to receiver.

Reply:
> not sure about the CALLBACK_NUMBYTES.
> Do you mean add `CALLBACK_NUMBYTES: constant(Bytes[4096])` as a contract constant?

```vyper
CALLBACK_NUMBYTES: constant(uint256) = 4096

Reply:
> 

Removed the receiver part. Thank you for pointing it out!

Reply:
> > not sure about the CALLBACK_NUMBYTES.
> > Do you mean add `CALLBACK_NUMBYTES: constant(Bytes[4096])` as a contract constant?
> 
> ```vyper
> CALLBACK_NUMBYTES: constant(uint256) = 4096
> ```

added the constant, applied it to the interface